### PR TITLE
Rabin Karp in C

### PR DESCRIPTION
**Pull request by disha2sinha**
Fixes #2161 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by disha2sinha** [comment]
Changes requested in PR #2308 is made in this PR

File: Rabin_Karp/Rabin_Karp.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#include <stdio.h>
+#include <string.h>
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.c]: keep a line space after this

File: Rabin_Karp/Rabin_Karp.c [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+void Rabin_Karp ( char *text , char *pattern )
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.c]: remove the pace before comma

File: Rabin_Karp/Rabin_Karp.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+int main()
+{
+    char text [30] ;
+    char pattern [15] ;
+    printf ( "Enter Text:\n" ) ;
+    scanf ( "%s",text ) ;
+    printf ( "Enter Pattern to be searched:\n" ) ;
+    scanf ( "%s", pattern ) ;
+    Rabin_Karp ( text , pattern ) ;
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.c]: keep space after comma
don't keep space before the comma

File: Rabin_Karp/Rabin_Karp.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    return 0;
+}
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.c]: keep a line space after this

File: Rabin_Karp/Rabin_Karp.c [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+/* Input/Output Demo:
+Enter Text:
+bbaavcaavaavifjkdfkfaav
+Enter Pattern to be searched:
+aav
+
+
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.c]: remove this lines

File: Rabin_Karp/Rabin_Karp.c [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+Pattern searched is found at position 3
+
+
+Pattern searched is found at position 7
+
+
+Pattern searched is found at position 10
+
+
+Pattern searched is found at position 21*/
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.c]: remove the unnecessary lines

**Comment by asha15** [comment]
keep a line space after variable declaration and before loops

**Comment by disha2sinha** [comment]
Changes requested has been made..

File: Rabin_Karp/Rabin_Karp.c [hunk 7]
```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <string.h>
+
+void Rabin_Karp (char *text, char *pattern)
+{
+    int text_len = strlen (text) ;
```
Comment (SakshiShreya) [inline Rabin_Karp/Rabin_Karp.c]: Remove the space before semicolon.

File: Rabin_Karp/Rabin_Karp.c [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        hash_pat = (hash_pat * 256 + pattern [i]) % 149; // hash value for pattern
+    }
+    
+    for(int i = 0 ; i < = text_len - pattern_len ; i++)
+    {
+        int j;
+        
+        if (hash_txt == hash_pat) 
+        {
+
```
Comment (somya-kapoor) [inline Rabin_Karp/Rabin_Karp.c:29]: Remove this line

File: Rabin_Karp/Rabin_Karp.c [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    char text [30];
+    char pattern [15];
+    printf ("Enter Text:\n");
+    scanf ("%s", text);
+    printf ("Enter Pattern to be searched:\n");
+    scanf ("%s", pattern);
+    Rabin_Karp (text, pattern);
+    return 0;
+}
+
```
Comment (somya-kapoor) [inline Rabin_Karp/Rabin_Karp.c:64]: Remove this line

**Comment by disha2sinha** [comment]
changes made

**Comment by disha2sinha** [comment]
changes made

File: Rabin_Karp/Rabin_Karp.c [hunk 10]
```diff
@@ -0,0 +1,72 @@
+#include <stdio.h>
+#include <string.h>
+
+void Rabin_Karp (char *text, char *pattern)
+{
+    int text_len = strlen (text);
+    int pattern_len = strlen (pattern);
+    int hash = 1;
+    
+    for (int i = 0 ; i < pattern_len - 1 ; i++)
```
Comment (somya-kapoor) [inline Rabin_Karp/Rabin_Karp.c:10]: Remove space before ;
for (int i = 0; i < pattern_len - 1; i++)

File: Rabin_Karp/Rabin_Karp.c [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        hash = (hash * 256) % 149;
+        // there are 256 total characters in Ascii table & 149 is a prime number chosen randomly
+    }
+    int hash_txt = 0, hash_pat = 0;
+    
+    for (int i = 0 ; i < pattern_len ; i++)
```
Comment (somya-kapoor) [inline Rabin_Karp/Rabin_Karp.c:17]: Remove space before ;

File: Rabin_Karp/Rabin_Karp.c [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    
+    for(int i = 0 ; i < = text_len - pattern_len ; i++)
+    {
+        int j;
+        
+        if (hash_txt == hash_pat) 
+        {
+            // checking if all the chars are equal in text portion & pattern
+            for (j = 0 ; j < pattern_len ; j++)
```
Comment (somya-kapoor) [inline Rabin_Karp/Rabin_Karp.c:30]: Remove space before ;

**Comment by disha2sinha** [comment]
Changes made

**Comment by somya-kapoor** [comment]
Please squash the commits, then it will be ready to be merged.

**Comment by disha2sinha** [comment]
> 
> 
> Please squash the commits, then it will be ready to be merged.

**Comment by disha2sinha** [comment]
> 
> 
> > Please squash the commits, then it will be ready to be merged.

Due to some errors, I couldn't squash the commits...so made a seperate PR #2844

**Pull request closed without merging**